### PR TITLE
Achievements system improvement

### PR DESCRIPTION
**Pull request by alisonrag**
improvement of Achievements system

- added achievement_list file
- updated achievement related packets
- updated "achieve" command

screenshot:
achievements

### CODE REVIEW

**Comment by alisonrag** [comment]
reward that give item, buff and title:
image

**Comment by ya4ept** [comment]
may be `achive desc <achievementID>`  ?

File: src/Commands.pm [hunk 1]
```diff
@@ -7033,6 +7033,29 @@ sub cmdAchieve {
 			$messageSender->sendAchievementGetReward($args[1]);
 		}
 
+	} elsif ($args[0] =~ /^\d+$/) {
+		if(defined($achievements{$args[0]})) {
+			# status
+			message center(" " . T("Achievement Info") . " ", 43, '-') . "\n";
```
Comment (ya4ept) [inline src/Commands.pm:7039]: ~~43~~ -> 79

File: src/Commands.pm [hunk 2]
```diff
@@ -7033,6 +7033,29 @@ sub cmdAchieve {
...
+			message TF("Buff: %s\n", ($achievements{$args[0]}->{rewards}->{buff}) ? $statusName{$statusHandle{$achievements{$args[0]}->{rewards}->{buff}}} : "N/A");
+			message TF("Title: %s\n", ($achievements{$args[0]}->{rewards}->{title}) ? $title_lut{$achievements{$args[0]}->{rewards}->{title}} : "N/A");
+			message T("Status:\n");
+			if ( defined ( $achievementList->{$args[0]} ) ) {
+				my $achieve = $achievementList->{$args[0]};
+				message TF("%s %s\n", $achieve->{completed} ? T("complete") : T("incomplete"), $achieve->{reward}  ? T("rewarded") : T("not rewarded"));
+			} else {
+				message T("N/A\n");
+			}
+			message center("", 43, '-') . "\n";
```
Comment (ya4ept) [inline src/Commands.pm:7055]: ~~43~~ -> 79

File: src/Commands.pm (hunk 1 above)
Comment (ya4ept) [inline src/Commands.pm:7039]: this is a bad design:
```
message "***";
message "***";
message "***";
```
see [quest info](https://github.com/OpenKore/openkore/blob/eb1af8dbf59655dec5517e51331c8de46da14d20/src/Commands.pm#L6806) :
```
my $msg = "***";
$msg .= "***";
message $msg;
```

File: src/Commands.pm [hunk 3]
```diff
@@ -7033,6 +7033,29 @@ sub cmdAchieve {
...
+		} else {
+			warning T("Achievement not Found, consider update the achievement_list.txt\n");
```
Comment (ya4ept) [inline src/Commands.pm:7057]: may be: `The achievement was not found. Update the 'achievement_list.txt' file`

**Review by ya4ept — CHANGES_REQUESTED** [review]

File: src/Commands.pm (hunk 1 above)
Comment (alisonrag) [reply src/Commands.pm]: fixed at https://github.com/OpenKore/openkore/pull/3288/commits/b70aa86d970483a95b4f1b79a2ed46c7f9c756b5

File: src/Commands.pm (hunk 2 above)
Comment (alisonrag) [reply src/Commands.pm]: fixed at https://github.com/OpenKore/openkore/pull/3288/commits/b70aa86d970483a95b4f1b79a2ed46c7f9c756b5

File: src/Commands.pm (hunk 3 above)
Comment (alisonrag) [reply src/Commands.pm]: fixed at https://github.com/OpenKore/openkore/pull/3288/commits/b70aa86d970483a95b4f1b79a2ed46c7f9c756b5

**Comment by alisonrag** [comment]
> may be:
> 
> 1. replace the command name with `achive desc <achievementID>`  or `achive info <achievementID>`?
> 2. rename file achievement_list.txt to achievement_list.lua?

1 - updated to achieve info <achievementID>
2 - its better keep it in .txt, cause original lua have functions, this '.txt' file have only the achievement table

**Comment by alisonrag** [comment]
image

**Comment by ya4ept** [comment]
изображение

**Comment by ya4ept** [comment]
can we make commands?
```
achieve info <achievement #>
chieve reward <achievement #>
```

**Comment by alisonrag** [comment]
Nice update, i think is better use achieve id in both commands

**Comment by ya4ept** [comment]
> Nice update, i think is better use achieve id in both commands

it's a very long time to write :)

**Review by ya4ept — APPROVED** [review]

**Pull request merged**